### DISCUSSION TITLE
MD5 Index for Non Supported Sets (Notification on loading ROM, ROM Forwarding)

### DISCUSSION THREAD

**Discussion by ghost** [RetroAchievements/RAWeb#781] [Ideas]
As RetroAchievements evolves and more users will be using the Service in the future, it would be nice to have a Notification System on loading a ROM and ROM Forwarding Index of (current, or never) Non Supported MD5s instead of getting the 'No Achievements Found' message.

Not everyone is browsing the Site, looking into the Forums to get the correct version of the Game which is linked to a Set. To eliminate the issue of finding the correct ROM or at least reduce is drastically, we could have a MD5 List for those other ROMS which aren't supported due to being Regional Difference, Wrong Revision, Alternative Dump etc.

**Example:**
A User wants to play a Game with Achievements ON but don't know which Version to choose, the MD5 Index of Non Supported ROMs can be used to Notify the User that the loaded Game isn't Supported yet (will be displayed in the Emulators as Popup Message) and give him a choice to forward to the Version which is supported.

| MD5 | Source | Status | Notification | Forward-To MD5 | Forward-To GameID |
| :---: | :---: | :---: | :---: | :---: |  :---: | 
| 5A5C2F4F1CAFB1F55A8DC0D5AD4550E5 | Primary | Supported |  |  | |
| 534045984318FDF989CD1D15DC8D1664 | Secondary | Supported | Forward to Higher Compatible Version? | 5A5C2F4F1CAFB1F55A8DC0D5AD4550E5 | 1447 |
| D306C54CCFDF5CB4F8EC588F19B3E33D | Secondary | Unsupported | Forward to Correct Version? | 5A5C2F4F1CAFB1F55A8DC0D5AD4550E5 | 1447 |

**MD5:**
This is the linked MD5 on the Site, the Primary would be the Version with highest compatibility (used as source for developing Achievements).

**Source:**
Primary - Highest Compatibility
Secondary - Alternative linked Versions (Regions, Translations etc.)

**Status:**
Current Status of the MD5 (could be turned ON/OFF)

**Notification:**
This is the Message User will see when they load the Game (MD5) in a Emulator.
- Loading a Supported Version can give the User a Choice to Forward to the higher compatible version (This is for Language differences).
- Loading a Unsupported Version could be Auto-Forwarded (option in Emulators) to the correct version.

**Forward-To MD5:**
This is the MD5 the Emulator will forward to the Higher Compatible/Correct Version.

- RetroArch has a creating Playlist feature, which indexed MD5s (based on scanned content on users Harddrive) could be used to Auto-Forward to the correct Version.

- RetroArch could also give the User all sort of functions such as:

1. Turn Version Forwarding ON/OFF (for Unsuported Versions)
2. Always choose Primary Source (for Supported Versions)
3. Ask before loading a Game (This would display the Notificaton Message)

**Forward-To GameID:**
This is to load the Achievements for the loaded Game.

**Final Note**

This is just a small example how to make loading the correct version a lot easier for those Users who don't visit the Site/Forums that often and look for compatible MD5s.

The biggest impact would be the RetroArch support here as it allows to scan the content on your Harddrive and creates a content index, which would turn this whole process of almost full automatic. Alternatively for those who don't have a playlist created can benefit from the Nofitication Message on loading a Game, instead of the 'No Achievements Found' message which leaves many in the dark.

I decided to create a new issue regards this as its kinda its own thing, but it could be added to the Database Manager https://github.com/RetroAchievements/RAWeb/issues/133.

**Comment by Jamiras** [comment]
This sort of information would have to be stored in the hash list (see also #717)

**Comment by Jamiras** [comment]
#3404 allows hashes for non-supported ROMs to indicate their incompatibility to the client.

I'm not sure if I understand the forwarding mechanic, but it sounds like we would need to know what other ROMs the client has available, and I don't think that's something we want to try to manage.
